ui/src/SensitiveData.js

```diff
@@ -0,0 +1,161 @@
+import React, { Component } from "react";
+import "@fontsource/roboto";
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+} from "@material-ui/core";
+
+
+class SensitiveData extends Component {
+  state = {
+    columnToUpdate: "",
+    newValue: "",
+    newValueValidationError: "",
+    showDialog: false,
+    allowableSensitiveDataColumns: []
+  };
+
+  openDialog = () => {
```

Comment:
I would combine `refreshDataFromBackend` into this method and not show the dialog until the data has arrived from the backend - single state update